webapp/app.py

```diff
@@ -1249,15 +1249,13 @@ def render_supermicro_blogs():
 def render_form(form):
     @wraps(render_form)
     def wrapper_func():
-        with app.app_context() and app.test_request_context():
-            return flask.render_template(
-                form["templatePath"],
-                fieldsets=form["fieldsets"],
-                formData=form["formData"],
-                isModal=form.get("isModal"),
-                modalId=form.get("modalId"),
-            )
-
+        return flask.render_template(
+            form["templatePath"],
+            fieldsets=form["fieldsets"],
+            formData=form["formData"],
+            isModal=form.get("isModal"),
+            modalId=form.get("modalId"),
+        )
```

Comment:
`set_form_rules()` is called on line 1270. Is there a way to only call the form rule if the form template is present? It feels a bit redundant to be calling this function on pages that are not using it.

Reply:
The `set_form_rules()` runs on project build, not individual pages. We could check for the presence of the template hook within the document, but this gets complicated as the template can exist within an extend, such as `base_desktop.html`. Instead I have added some error handling to catch these cases, we can then remove them from the form data manually.